### PR TITLE
Week01 submissions

### PR DESCRIPTION
**Pull request by AhsanShafiqShawon**
Please review my code

### CODE REVIEW

File: week01/LongestConsecutiveSequence.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Solution {
+public:
+    int longestConsecutive(vector<int>& nums) {
+        unordered_set<int> hashSet;
+        for(int num : nums) {
+            hashSet.insert(num);
+        }
```
Comment (Sadman007) [inline week01/LongestConsecutiveSequence.cpp:7]:
```suggestion
        unordered_set<int> hashSet (nums.begin(), nums.end());
```

File: week01/MergeIntervals.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Solution {
+public:
+    vector<vector<int>> merge(vector<vector<int>>& intervals) {
+        vector<vector<int>> result;
+        sort(intervals.begin(), intervals.end());
+        vector<int> temp = intervals[0];
+        if(intervals.size() == 0)
+            return result;
```
Comment (Sadman007) [inline week01/MergeIntervals.cpp:8]: If `intervals.size()` could ACTUALLY be 0, line no 6 would have resulted in Runtime Error. So, this `if` condition is redundant.

File: week01/MergeIntervals.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Solution {
+public:
+    vector<vector<int>> merge(vector<vector<int>>& intervals) {
+        vector<vector<int>> result;
+        sort(intervals.begin(), intervals.end());
+        vector<int> temp = intervals[0];
```
Comment (Sadman007) [inline week01/MergeIntervals.cpp:6]: `temp` is a very bad variable naming in tech interviews. Find something meaningful.

i.e. `vector<int> base = intervals[0];`

File: week01/ProductOfArrayExceptSelf.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+            prefix *= nums[i];
+            result[i+1] = prefix;
+        }
+        for(int i = n-1; i > 0; i--) {
+            postfix *= nums[i];
+            result[i-1] *= postfix;
+        }
+        return result;
+    }
+};
```
Comment (Sadman007) [inline week01/ProductOfArrayExceptSelf.cpp:19]: Complexity analysis?

File: week01/SingleNumber.cpp [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        }
+        return ans;
+    }
+};
+
+/*
+Time Complexity: O(N)
+Memory Complexity: O(1)
+Solution/Idea: This problem can be solved in many ways, but most optimal solution is in sense of memory and time complexity is XORing all the elements.
+Learning: a ^ a = 0
```
Comment (Sadman007) [inline week01/SingleNumber.cpp:16]: The most important learning from this problem should be `the commutative law of XOR operation`.

File: week01/LongestConsecutiveSequence.cpp (hunk 1 above)
Comment (AhsanShafiqShawon) [reply week01/LongestConsecutiveSequence.cpp:7]: Smart coding, thank you.

File: week01/MergeIntervals.cpp (hunk 2 above)
Comment (AhsanShafiqShawon) [reply week01/MergeIntervals.cpp:8]: Okay, go the point.

File: week01/MergeIntervals.cpp (hunk 3 above)
Comment (AhsanShafiqShawon) [reply week01/MergeIntervals.cpp:6]: Sure, I will be more concious about this.

File: week01/ProductOfArrayExceptSelf.cpp (hunk 4 above)
Comment (AhsanShafiqShawon) [reply week01/ProductOfArrayExceptSelf.cpp:19]: I will add shortly.

File: week01/SingleNumber.cpp (hunk 5 above)
Comment (AhsanShafiqShawon) [reply week01/SingleNumber.cpp:16]: Definitely

File: week01/MergeIntervals.cpp (hunk 3 above)
Comment (Sadman007) [reply week01/MergeIntervals.cpp:6]: You can change it now. Don't wait for the next weeks. Thanks.

**Review by Sadman007 — APPROVED** [review]
Hello @AhsanShafiqShawon 

Nice work.

File: week01/MergeIntervals.cpp (hunk 3 above)
Comment (AhsanShafiqShawon) [reply week01/MergeIntervals.cpp]: Done

**Pull request merged**